src/http/client.rs

```diff
@@ -166,49 +154,38 @@ impl<'a> HttpBuilder<'a> {
 
         Ok(self)
     }
-}
 
-impl<'a> Future for HttpBuilder<'a> {
-    type Output = Result<Http>;
+    /// Use the given configuration to build the `Http` client.
+    pub fn build(mut self) -> Http {
+        let token = self.token.take().unwrap();
```

Comment:
Since we are no longer behind `&mut self`, we can safely take ownership of each individual field. As a result, `Option::take` isn't necessary anymore.
```suggestion
        let token = self.token.unwrap();
```